finetrainers/trainer.py

```diff
@@ -741,14 +748,21 @@ def train(self) -> None:
                 transformer_lora_layers=transformer_lora_layers,
             )
 
+            if self.args.push_to_hub:
+                upload_folder(
+                    repo_id=self.state.repo_id, folder_path=self.args.output_dir, ignore_patterns=["checkpoint-*"]
+                )
+            
+            self.validate(step=global_step, final_validation=True)
```

Comment:
We should do a final validation as it helps to check the effectivity of `load_lora_weights()`.